### ISSUE TITLE
HTML preview broken

### ISSUE THREAD

**Issue by pcanterino** [wereturtle/ghostwriter#463]
I'm using ghostwriter 1.8.0 on Linux Mint 19.2 (VirtualBox 6.0.10).

The HTML preview is broken. Even with a simple Markdown document (only text, no formatting), the HTML preview only shows a black panel:

![black_panel](https://user-images.githubusercontent.com/11569243/64759842-8e12ea80-d538-11e9-9bc8-36aba35e849e.png)

It even renders the program unusable. The menus don't work, but I can type text. I can disable HTML preview using CTRL+M, but the menus still don't work, typing also doesn't work then (if I enable HTML preview using CTRL+M, I can type text again). I can't even close the program, I have to `kill` it. When I minimize the program, the content of the window stays on the desktop:

![minimized](https://user-images.githubusercontent.com/11569243/64759863-98cd7f80-d538-11e9-931b-9fbf6874a636.png)

After restarting program, it still doesn't work. After removing the config file *~/.config/ghostwriter/ghostwriter.conf*, the program works again.

On a similar VM (also Linux Mint 19.2, but a fresh install) on a different machine, ghostwriter works flawlessly.

Please let me know if you need further information.

**Comment by wereturtle** [comment]
It looks like you fixed your own problem.  It looks like maybe ghostwriter didn't have permission to write to it's own configuration file (or perhaps to read it)?

Let me know if the issue returns.  This the second weird report I've seen that had to do with the configuration file.  I'm wondering what's going on.

**Comment by pcanterino** [comment]
Sorry, my report was misleading.

If I delete the config file, the program works again. But as soon as I open HTML preview, the program becomes unusable again. Then I have to do the same procedure to get it working again.

**Comment by wereturtle** [comment]
What are the permissions on your `~/.config/ghostwriter` folder?

Though, I"m thinking it's more that when you close ghostwriter, it saves the information that your HTML preview pane is open, and thus when you relaunch ghostwriter after killing it, it goes right back to being spazzy with the HTML preview pane being opened automatically.

Also, did you install with Flatpak, the PPA or a private build?

**Comment by pcanterino** [comment]
I installed with PPA.

I will check the permissions of the folder as soon as I'm back at my office (perhaps on wednesday).

**Comment by pcanterino** [comment]
The permissions of *~/.config/ghostwriter* are set to 775.

**Comment by pcanterino** [comment]
From what I see if I try to close ghostwriter with the black and broken HTML preview and an unsaved document, the confirmation dialogue for saving appears **behind** the main window. If I move the broken main window, the confirmation dialogue appears.

Saving or the document or closing makes the whole screen become black. The window manager does not crash, it is just black. If I press the SUPER / WIN button, the desktop comes back.

**Comment by wereturtle** [comment]
OK, this will seem like a bizarre question, but it's something I noticed on Flatpak in the early days before they figured out their Nvidia + Chromium issues.  Do you have an Nvidia card, and are you using the Nouveau driver instead of the proprietary Nvidia driver?  I think Chromium (which is what ghostwriter is using under the hood for its HTML preview) needs proper video acceleration, and I'm not sure how it behaves with Nouveau.  It might be OK, but if you are using Nouveau, try switching to the Nvidia proprietary driver and see if your issue goes away.  I just remember strange things on Flatpak before they figured out their Nvidia driver stuff.

**Comment by pcanterino** [comment]
As I said, the problem occurs on VirtualBox.

But that was the right idea: As I already mentioned, the problem occurs on one VM, while on the other, ghostwriter works correctly. I compared the display settings and changed the graphics controller from *VBoxVGA* to *VMSVGA*. This fixed the problem.
*VMSVGA* is the default for new Linux VMs in newer versions VirtualBox. I migrated from an older version, that might be the origin of the wrong setting.

Turning of 3D Acceleration also "fixes" the problem, but Linux Mint does not recommend running itself in that way.

Thank you for your support, I will close this issue.
